### PR TITLE
Update to PHPUnit 8

### PR DESCRIPTION
**Pull request by C0pyR1ght**
Updated from PHPUnit version 7 -> 8

Codebase changes were needed:
- Updated Composer PHPUnit version to ^8
- Refactored test to work with PHPUnit8
- Added PHPUnit Cache to gitignore

!important: PHPUnit8 requires PHP ^7.2 -> PHP version has to be updated bevor merging.
because of that not all checks are successfull.

//cc @staabm

### CODE REVIEW

**Comment by staabm** [comment]
thx, looks good.

remaining todos:
- uplift min php requirement to 7.2
- drop old phpunit versions

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
         "sabre/uri"    : "^2.0"
     },
     "require-dev" : {
-        "phpunit/phpunit" : "^6.0 || ^7.0"
+        "phpunit/phpunit" : "^6.0 || ^7.0 || ^8.0"
```
Comment (staabm) [inline composer.json]: for 7.1+ we no longer need phpunit 6.x

```suggestion
        "phpunit/phpunit" : "^7.0 || ^8.0"
```

**Comment by staabm** [comment]
interessting.. the php 7.4 build uses phpunit 6.x.. it should be phpunit 8.x though

File: composer.json (hunk 1 above)
Comment (C0pyR1ght) [reply composer.json]: Did you intentionally remove the `^` on version 7?

**Comment by C0pyR1ght** [comment]
todos done, CI checks are successfull.
ready for merging.

@staabm

**Comment by staabm** [comment]
looks really good, thx

**Pull request merged**